static/flow/views/program-editor-block-palette.js

```diff
@@ -58,43 +58,104 @@ var ProgramEditorBlockPalette = function(options) {
     //
     // create the sensor menu section main button
     //
-    var createSensorBtn = function(name, type, index) {
+    var createBtn = function(name, type) {
+        
+        var div = $('<div>', { class: 'diagram-menu-entry container-light-gray' } );
         var btn;
-        if(index%2 == 0){
-            btn = $('<button>', { class: 'diagramMenuEntry menulightgray' } );
-        }
-        else{
-            btn = $('<button>', { class: 'diagramMenuEntry menudarkgray' } );
-        }
+        btn = $('<span>', { class: '' } );
 
+        //icon
+        var menuIcon = $('<img class="menu-icon">'); 
+        if(type=="temperature")
```

Comment:
I'd consider passing in the `filename` to this function so you can just construct the link and avoid this giant block

Reply:
Yeah, it's crude.  Wasn't sure if people are even going to accept the icons.  It's still up in the air if people like/want them.